lib/improver/wind_direction.py

```diff
@@ -0,0 +1,409 @@
+
+        Returns:
+            (np.ndarray or float):
+                3D array or float - wind direction translated to
+                complex numbers.
+
+        """
+
+        # Convert from degrees to radians.
+        angle_rad = np.deg2rad(angle_deg)
+
+        # Derive real and imaginary components (also known as a and b)
+        real = radius*np.cos(angle_rad)
+        imag = radius*np.sin(angle_rad)
+
+        # Combine components into a complex number and return.
+        return real + 1j*imag
+
+    @staticmethod
+    def complex_to_deg(complex_in):
+        """Converts complex to degrees.
+
+        The "np.angle" function returns negative numbers when the input
+        is greater than 180. Therefore additional processing is needed
+        to ensure that the angle is between 0-359.
+
+        Args:
+            complex_in (np.ndarray):
+                3D array - wind direction angles in
+                complex number form.
+
+        Returns:
+            angle (np.ndarray):
+                3D array - wind direction in angle form
+
+        Raises
+        ------
+        TypeError: If complex_in is not an array.
+
+        """
+
+        if not isinstance(complex_in, np.ndarray):
+            msg = "Input data is not a numpy array, but {}"
+            raise TypeError(msg.format(type(complex_in)))
+
+        angle = np.angle(complex_in, deg=True)
+
+        # If angle negative value - add 360 degrees.
+        angle = np.where(angle < 0, angle+360, angle)
+
+        # If angle == 360 - set to zero degrees.
+        # Due to floating point - need to round value before using
+        # equal operator.
+        round_angle = np.around(angle, 2)
+        angle = np.where(round_angle == 360, 0.0, angle)
+
+        return angle
+
+    @staticmethod
+    def find_r_values(complex_in):
+        """Find radius values from complex numbers.
+
+        Takes input wind direction in complex values and returns array
+        containing r values using Pythagoras theorem.
+
+        Args:
+            complex_in (np.ndarray or float):
+                3D array or float - wind direction angles in complex numbers.
+
+        Returns:
+            (np.ndarray or float):
+                3D array or float - Radius values extracts from
+                complex numbers.
+        """
+
+        return np.sqrt(np.square(complex_in.real)+np.square(complex_in.imag))
+
+    @staticmethod
+    def calc_polar_mean_std_dev(wind_dir_complex, wind_dir_deg_mean,
+                                r_vals, r_thresh, realization_axis):
+        """Find standard deviation of polar numbers.
+
+        The average wind direction complex values represent the midpoint
+        between the different values and so have r values between 0-1.
+
+        1) From wind_dir_deg_mean - create a new set of complex values.
+           Therefore they will have the same angle but r is fixed as r=1.
+        2) Use fixed-r average complex values to find the distance between the
```

Comment:
Find the distance between the mean point and all the ensemble member wind direction complex values